src/scripts/controller/productController.ts

```diff
@@ -0,0 +1,73 @@
+import { Product } from '../model/interfaces';
+import { debounce } from '../utils/debounce';
+import { ProductView } from '../view/productView';
+import { ApiServices } from './apiServices';
+
+export class ProductControler {
+    private apiServices: ApiServices;
+    private allProducts: Product[] = [];
+    private isLoading: boolean = true;
+    private view!: ProductView;
+
+    private searchBar: HTMLInputElement;
+
+    constructor() {
+        this.apiServices = new ApiServices();
+        this.view = new ProductView();
+
+        this.searchBar = document.getElementById(
+            'search-bar'
+        ) as HTMLInputElement;
+
+        this.initalize();
+    }
+
+    private async initalize(): Promise<void> {
```

Comment:
Nit.
Did you mean `initialize` ?
```suggestion
    private async initialize(): Promise<void> {
```

Reply:
Rename function name(https://github.com/krushna-simform/product-management-system/pull/3/commits/7007557a824ca530bdaa5a53df8427b0966f6a0b)